From 485731b73c98622b2e4b5fd48048ad73b7b1047b Mon Sep 17 00:00:00 2001
From: Nikolay Aleksandrov <nikolay@redhat.com>
Date: Mon, 1 Sep 2014 13:33:21 -0400
Subject: [crypto] qat: Use pci_enable_msix_exact() instead of pci_enable_msix()

Message-id: <1409578401-26943-30-git-send-email-nikolay@redhat.com>
Patchwork-id: 90106
O-Subject: [RHEL7.1 PATCH BZ 1091130 29/29] crypto: qat - Use pci_enable_msix_exact() instead of pci_enable_msix()
Bugzilla: 1091130
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Herbert Xu <herbert.xu@redhat.com>

From: Alexander Gordeev <agordeev@redhat.com>

As result of deprecation of MSI-X/MSI enablement functions
pci_enable_msix() and pci_enable_msi_block() all drivers
using these two interfaces need to be updated to use the
new pci_enable_msi_range()  or pci_enable_msi_exact()
and pci_enable_msix_range() or pci_enable_msix_exact()
interfaces.

Cc: Tadeusz Struk <tadeusz.struk@intel.com>
Cc: Herbert Xu <herbert@gondor.apana.org.au>
Cc: "David S. Miller" <davem@davemloft.net>
Cc: qat-linux@intel.com
Cc: linux-crypto@vger.kernel.org
Cc: linux-pci@vger.kernel.org
Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
(cherry picked from commit 51269ad2880774e7d46fb1e0edb7a83e15d05d33)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/crypto/qat/qat_dh895xcc/adf_isr.c b/drivers/crypto/qat/qat_dh895xcc/adf_isr.c
index d4172de..1082e3b 100644
--- a/drivers/crypto/qat/qat_dh895xcc/adf_isr.c
+++ b/drivers/crypto/qat/qat_dh895xcc/adf_isr.c
@@ -70,9 +70,9 @@ static int adf_enable_msix(struct adf_accel_dev *accel_dev)
  for (i = 0; i < msix_num_entries; i++)
   pci_dev_info->msix_entries.entries[i].entry = i;
 
- if (pci_enable_msix(pci_dev_info->pci_dev,
-       pci_dev_info->msix_entries.entries,
-       msix_num_entries)) {
+ if (pci_enable_msix_exact(pci_dev_info->pci_dev,
+      pci_dev_info->msix_entries.entries,
+      msix_num_entries)) {
   pr_err("QAT: Failed to enable MSIX IRQ\n");
   return -EFAULT;
  }
-- 
1.7.1